lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscovery/RandomSpec.hs

```diff
@@ -0,0 +1,244 @@
+        , addr =
+                "82d818584283581cb039e80866203e82fc834b8e6a355b83ec6f8fd199\
+                \66078a40e6d6b2a101581e581c22e27fb12d08728073cd416dfbfcb8dc\
+                \0e760335d1d60f65e8740034001a4bce4d1a"
+        , accIndex =
+                2694138340
+        , addrIndex =
+                2512821145
+        , expected = True
+        }
+    it "check isOurs for bogus address" $
+        checkIsOurs CheckIsOursTest
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584283581cb039e80866203e82fc834b8e6a355b83ec6f8fd199"
+        , accIndex =
+                2694138340
+        , addrIndex =
+                2512821145
+        , expected = False
+        }
+
+goldenSpecTestnet :: Spec
+goldenSpecTestnet =
+    describe "Golden tests forByron Addresses w/ random scheme (Testnet)" $ do
+    it "check isOurs - initial account" $
+        checkIsOurs CheckIsOursTest
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584983581ca03d42af673855aabcef3059e21c37235ae706072d\
+                \38150dcefae9c6a201581e581c22e25f2464ec7295b556d86d0ec33bc1\
+                \a681e7656da92dbc0582f5e402451a4170cb17001a39a0b7b5"
+        , accIndex =
+                2147483648
+        , addrIndex =
+                2147483648
+        , expected = True
+        }
+    it "check isOurs - another account" $
+        checkIsOurs CheckIsOursTest
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584983581c267b40902921c3afd73926a83a23ca08ae9626a64a\
+                \4b5616d14d6709a201581e581c22e219c90fb572d565134f6daeab650d\
+                \c871d130430afe594116f1ae02451a4170cb17001aee75f28a"
+        , accIndex =
+                3337448281
+        , addrIndex =
+                3234874775
+        , expected = True
+        }
+
+{-------------------------------------------------------------------------------
+                    Golden tests for Address derivation path
+-------------------------------------------------------------------------------}
+
+data CheckIsOursTest = CheckIsOursTest
+    { mnem :: [Text]
+    , addr :: ByteString
+    , accIndex :: Word32
+    , addrIndex :: Word32
+    , expected :: Bool
+    } deriving (Show, Eq)
+
+-- An arbitrary mnemonic sentence for the tests
+arbitraryMnemonic :: [Text]
+arbitraryMnemonic =
+    [ "price", "whip", "bottom", "execute", "resist", "library"
+    , "entire", "purse", "assist", "clock", "still", "noble" ]
+
+-- A different arbitrary mnemonic sentence for the tests.
+anotherMnemonic :: [Text]
+anotherMnemonic =
+    [ "toss", "reunion", "lunar", "pilot", "direct", "chicken"
+    , "give", "total", "future", "wrap", "sunny", "ostrich" ]
+
+checkIsOurs :: CheckIsOursTest -> Expectation
+checkIsOurs CheckIsOursTest{..} = do
+    fst (isOurs addr' rndState) `shouldBe` expected
+    fst (isOurs addr' rndStateWrong) `shouldBe` False
```

Comment:
This assertion is slightly awkward. Seems like the test is doing two things at once.
In general, I'd rather have this sort of things done in separate tests, especially when the structure of the test already allows for testing such case.